### PR TITLE
GetDeletedSecret API

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 1]
```diff
@@ -10,7 +10,8 @@
 #include "../src/private/keyvault_protocol.hpp"
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:10]: This is an scape. Public headers shall not include private headers.  Make the required changes to only include the private headers from within the cpp files

**Review by vhvb1989 — COMMENTED** [review]
A question before approving

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp (hunk 1 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:10]: can't do that here , i need the protocol header for the definition of m_protocol member of the client class

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp [hunk 2]
```diff
@@ -48,7 +48,7 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
      * @param name The name of the secret.
      * @param value The name of the secret.
      */
-    KeyVaultSecret(std::string name, std::string value)
+    KeyVaultSecret(std::string name, std::string value = "")
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp]: ok, resolved this

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp [hunk 3]
```diff
@@ -50,11 +50,28 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
...
+  private:
+    KeyVaultSecret(std::string name) : Name(std::move(name))
     {
       if (Name.empty())
       {
         throw std::invalid_argument("Name cannot be empty");
       }
     }
+
+    friend struct KeyVaultDeletedSecret;
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp:74]: I want to know what @RickWinter will say here :)  .  He's something against the friend-strategy

**Review by vhvb1989 — APPROVED** [review]
I am fine and good with this.  But, just confirm with @RickWinter if he's fine having the friend-relation over there.

**Pull request merged**